packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -626,6 +633,17 @@ function flush_queued_effects(effects) {
 				current_batch.current.size > n &&
 				(effect.f & USER_EFFECT) !== 0
 			) {
+				var execution_count = (effect_execution_count.get(effect) ?? 0) + 1;
+				effect_execution_count.set(effect, execution_count);
+
+				// If many effects are executed, they cause another flush loop, which could
+				// lead to false-positives in the infinite loop detection. Therefore decrease
+				// the counter unless the individual effect has been executed many times, which
+				// indeed hints at an infinite loop.
+				if (execution_count < 1000) {
```

Comment:
Does this additional measure make sense to prevent a case where effects are constantly created/destroyed because you're playing ping-pong with an if block appearing/disappearing? Dunno if even possible, but maybe?
```suggestion
				if (execution_count < 1000 && effect_execution_count.size < 10_000_000) {
```